Sources/Rego/Engine.swift

```diff
@@ -48,16 +48,19 @@ extension OPA.Engine {
         ///   - tracer: (optional) The tracer to use for this evaluation.
         ///   - strictBuiltins: (optional) Whether to run in strict builtin evaluation mode.
         ///                     In strict mode, builtin errors abort evaluation, rather than returning undefined.
+        ///   - builtins: (optional) BuiltinRegistry to use for evaluation.
+        ///               When not provided, defaultRegistry will be used
         public func evaluate(
             input: AST.RegoValue = .undefined,
             tracer: OPA.Trace.QueryTracer? = nil,
-            strictBuiltins: Bool = false
+            strictBuiltins: Bool = false,
+            builtins: BuiltinRegistry? = nil
```

Comment:
@patrick-east you may have had thoughts on how to shape this interface in the context of the capabilities APIs you were looking at?